projects/front_power_distribution/src/lights_signal_fsm.c

```diff
@@ -0,0 +1,110 @@
+#include "lights_signal_fsm.h"
+#include "front_power_distribution_events.h"
+
+FSM_DECLARE_STATE(state_none);                 // no lights active
+FSM_DECLARE_STATE(state_left_signal);          // left signal active
+FSM_DECLARE_STATE(state_right_signal);         // right signal active
+FSM_DECLARE_STATE(state_hazard_signal);        // hazard active
+FSM_DECLARE_STATE(state_hazard_left_signal);   // hazard active and left signal active
+FSM_DECLARE_STATE(state_hazard_right_signal);  // hazard active and right signal active
+
+static bool prv_guard_on(const Fsm *fsm, const Event *e, void *context) {
+  return e->data == 1;
+}
+
+static bool prv_guard_off(const Fsm *fsm, const Event *e, void *context) {
+  return e->data == 0;
+}
+
+FSM_STATE_TRANSITION(state_none) {
+  SignalFsmStorage *storage = fsm->context;
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_left_event, &prv_guard_on, state_left_signal);
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_right_event, &prv_guard_on, state_right_signal);
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_hazard_event, &prv_guard_on, state_hazard_signal);
+}
+
+FSM_STATE_TRANSITION(state_left_signal) {
+  SignalFsmStorage *storage = fsm->context;
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_left_event, &prv_guard_off, state_none);
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_right_event, &prv_guard_on, state_right_signal);
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_hazard_event, &prv_guard_on, state_hazard_left_signal);
+}
+
+FSM_STATE_TRANSITION(state_right_signal) {
+  SignalFsmStorage *storage = fsm->context;
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_left_event, &prv_guard_on, state_left_signal);
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_right_event, &prv_guard_off, state_none);
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_hazard_event, &prv_guard_on,
+                             state_hazard_right_signal);
+}
+
+FSM_STATE_TRANSITION(state_hazard_signal) {
+  SignalFsmStorage *storage = fsm->context;
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_left_event, &prv_guard_on, state_hazard_left_signal);
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_right_event, &prv_guard_on, state_hazard_right_signal);
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_hazard_event, &prv_guard_off, state_none);
+}
+
+FSM_STATE_TRANSITION(state_hazard_left_signal) {
+  SignalFsmStorage *storage = fsm->context;
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_left_event, &prv_guard_off, state_hazard_signal);
+  FSM_ADD_GUARDED_TRANSITION(storage->signal_right_event, &prv_guard_on, state_hazard_right_signal);
```

Comment:
I think first left will be turned off, then this transitions to hazard, then right will be turned on, then we'll go to hazard right... so this won't happen either.